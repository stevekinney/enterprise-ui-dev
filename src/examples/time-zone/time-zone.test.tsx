import { test, expect, vi, afterEach, beforeEach } from 'vitest';
import { render } from 'test/utilities';
import TimeZone from '.';

beforeEach(() => {
  vi.useFakeTimers();
  vi.setSystemTime(1679492355195);
  // vi.setSystemTime(new Date(2012,9,19));
});

afterEach(() => {
  vi.useRealTimers();
});

/*
beforeAll(() => {
  vi.useFakeTimers();
  vi.setSystemTime(1679492355195);
});
*/

test('it should render successfully', () => {
  render(<TimeZone />);
});

test('should match the snapshot', async () => {
  const { container } = render(<TimeZone />);
  expect(container).toMatchSnapshot();
});
