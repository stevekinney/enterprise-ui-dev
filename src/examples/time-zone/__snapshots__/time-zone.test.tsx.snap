// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should match the snapshot 1`] = `
<div>
  <section
    class="flex flex-col gap-8 rounded-lg border-4 border-primary-50 p-8"
  >
    <h1
      data-testid="current-time"
    >
      1679492355195
    </h1>
    <p
      data-testid="prose"
    >
      You've been staring at this page for 
      0
       second(s).
    </p>
  </section>
</div>
`;
